azurerm/internal/services/network/firewall_resource.go

```diff
@@ -467,6 +497,47 @@ func flattenArmFirewallIPConfigurations(input *[]network.AzureFirewallIPConfigur
 	return result
 }
 
+func expandArmFirewallDNSSetting(input []interface{}) map[string]*string {
+	if len(input) == 0 {
+		return nil
+	}
+
+	v := input[0].(map[string]interface{})
+
+	var servers []string
+	for _, server := range v["servers"].([]interface{}) {
+		servers = append(servers, server.(string))
+	}
+
+	return map[string]*string{
+		"Network.DNS.EnableProxy": utils.String(fmt.Sprintf("%t", v["enabled"].(bool))),
```

Comment:
presumably this is a String to handle the Tri-state (True, False, Unset) here - is this defaulted/planned to be defaulted going forward? In general we're removing "enabled" fields and using the presence/omission of the block to infer that - but that won't work for tri-state fields, can we reach out to the service team and confirm the intention here?

Reply:
I will reply here later once I got the update from service team.

Reply:
any update @magodo ?

Reply:
@katbyte Unfortunately no update yet, I'll let you know once I got any news.

Reply:
Got the updates that `false` here is identical to absent. So I guess we shall remove the `Network.DNS.EnableProxy` and regard the precense of `Network.DNS.Servers` as an indication to whether enable this or not.

Reply:
@magodo no, it is not. Firewall supports DNS Proxy to "nowhere" aka Azure default dns.